src/eos-updater-util.c

```diff
@@ -367,14 +367,18 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
   shuffle_out_values (upgrade_ref, p_ref, NULL);
 
 out:
-  if ((p_ref || on_hold) && !metric_sent)
+  if (( (upgrade_ref && *upgrade_ref) || on_hold) && !metric_sent)
```

Comment:
nit: rm the space between the second and third open parentheses per https://github.com/endlessm/eos-documentation/wiki/C-Coding-Style#whitespace.